clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -223,9 +223,25 @@ public void connect(String id, InetSocketAddress address, int sendBufferSize, in
     /**
      * Register the nioSelector with an existing channel
      * Use this on server-side, when a connection is accepted by a different thread but processed by the Selector
-     * Note that we are not checking if the connection id is valid - since the connection already exists
+     * <p>
+     * If a connection already exists with the same connection id, the existing zombie connection is closed.
+     * These disconnections are notified and may be processed by the caller using {@link #disconnected()} before
+     * the next {@link #poll(long)}. Kafka brokers add an incrementing index to the connection id to avoid
+     * connection ids being reused even when remote port is reused, so this scenario is highly unlikely.
```

Comment:
@ijuma Thank you for the review.

No, counter per `Processor` is fine, since the issue is only if the reused id is registered with the same `Selector`. It is hard to say `impossible` when the only thing preventing it is a counter. But `int` should be big enough to avoid it altogether in practice.

The reason I handled this case rather than throw an exception is because an exception would result in the new connection being closed while the correct thing to do is to close the old connection, which was almost definitely closed by the remote-side and exists only because the broker hasn't yet cleared it. But even though I wrote the code to close the old channel, I would also prefer to just throw an exception and avoid the extra code. Personally I would leave the index as an `int` rather than a `long`, because if we hit the issue with `int`, it is more likely to be a resource leak where a channel is never closed rather than a timing issue with port reuse that the index is attempting to fix.